wml_aux/map2html/map2html.1
.\" Automatically generated by Pod::Man 2.28 (Pod::Simple 3.28)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{
.    if \nF \{
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "MAP2HTML 1"
.TH MAP2HTML 1 "2015-06-24" "EN Tools" "EN Tools"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
map2html \- convert server\-side to client\-side imagemap
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\&\fBmap2html\fR
[\fB\-t\fR \fItype\fR]
[\fB\-d\fR \fIshape\fR]
[\fB\-n\fR \fIname\fR]
[\fB\-o\fR \fIoutputfile\fR]
[\fB\-v\fR]
[\fIinputfile\fR]
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
The \fImap2html\fR program reads \fIinputfile\fR and converts
its contents from a server-side map to a client-side map.
.SH "OPTIONS"
.IX Header "OPTIONS"
.IP "\fB\-t\fR \fItype\fR" 4
.IX Item "-t type"
Sets the type of the server-side map. Either \f(CW\*(C`ncsa\*(C'\fR (default) or \f(CW\*(C`cern\*(C'\fR.
.IP "\fB\-d\fR \fIshape\fR" 4
.IX Item "-d shape"
Sets the default area shape. Either \f(CW\*(C`def\*(C'\fR (default) or \f(CW\*(C`rect\*(C'\fR.
.IP "\fB\-n\fR \fIname\fR" 4
.IX Item "-n name"
Produce a map called \fIname\fR. Default is \fIinputfile\fR.
.IP "\fB\-o\fR \fIoutputfile\fR" 4
.IX Item "-o outputfile"
This redirects the output to \fIoutputfile\fR. Usually the output will be send to
\&\f(CW\*(C`stdout\*(C'\fR if no such option is specified or \fIoutputfile\fR is "\f(CW\*(C`\-\*(C'\fR".
.IP "\fB\-v\fR" 4
.IX Item "-v"
This sets verbose mode where some processing information will be given on the
console.
.SH "AUTHOR"
.IX Header "AUTHOR"
.Vb 3
\& Ralf S. Engelschall
\& rse@engelschall.com
\& www.engelschall.com
.Ve


wml_aux/map2html/map2html.pl
#!/usr/bin/perl
eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
    if $running_under_some_shell;
##
##  map2html -- convert server-side to client-side imagemap 
##  Copyright (c) 1996-1998,1999 Ralf S. Engelschall, All Rights Reserved. 
##

require 5.003;

use lib "/usr/local/lib/wml/perl/lib";
use lib "/usr/local/lib/wml/perl/lib/x86_64-linux";

use Getopt::Long;

#
#   process command line
#
sub usage {
    print STDERR "Usage: map2html [options] mapfile\n";
    print STDERR "   where options are\n";
    print STDERR "   -t type  map format: 'ncsa' or 'cern' (default)\n";
    print STDERR "   -d shape default area shape: 'def' (default) or 'rect'\n";
    print STDERR "   -n name  produce a map called <name> (default is filename)\n";
    print STDERR "   -o file  output file\n";
    print STDERR "   -h       help (this text)\n";
    print STDERR "   -v       version information\n";
    exit(1);
}
sub version {
    print STDERR "This is MAP2HTML, Version 1.1\n";
    exit(0);
}
$opt_v = 0;
$opt_h = 0;
$opt_o = "-";
$opt_t = "ncsa";
$opt_d = "def";
$opt_n = "";
$Getopt::Long::bundling = 1;
$Getopt::Long::getopt_compat = 0;
$rc = Getopt::Long::GetOptions("v|verbose", 
                               "t|type=s", 
                               "d|defshape=s", 
                               "n|name=s",
                               "o|outputfile=s",
                               "h|help");
if (not $rc or $opt_h or ($#ARGV == -1)) {
    &usage;
}
if ($opt_v) {
    &version;
}

$server    = $opt_t;
$defshape  = $opt_d;
$name      = $opt_n;
$default   = "";
$lastc     = "";
$alt       = "";
  
if ($opt_o eq "-") {
    *OUT = *STDOUT;
}
else {
    open(OUT, ">$opt_o") || die "Cannot open output file ``$opt_o''";
}

for $file (@ARGV) {
    &domap(*OUT, $file, $name);
}

if ($opt_o ne "-") {
    close(OUT);
}

sub domap {
    local(*OUT, $file, $name) = @_;
    local(*IN);
    
    open(IN, "<$file") || warn "Can't open: $file\n", return;
    if ($name eq "") {
        $name = $file;
    }
    print OUT "<MAP NAME=\"$name\">\n"; # header
    while(<IN>) {
        &doline(*OUT, $_);
    }
    print OUT "$default"; # the default URL must be last
    print OUT "</MAP>\n";
    close(IN);
    return;
}

sub doline {
    local(*OUT, $line) = @_;
    local(@coords);
    local($coords, $href, $type);
  
    $type='dodgy'; # make sure that it doesn't print anything unless it's set
    $line =~ s|\n$||;
    $line =~ s|$||;
    if ($server =~ m|cern|i) {
        if ($line =~ m|^\s*default\s+(\S+)\s*$|i) {
            if ($defshape eq "rect") {
                $default = "<AREA SHAPE=RECT COORDS=\"0,0,9999,9999\" HREF=\"$1\" ALT=\"DEFAULT\">\n";
            }
            else {
                $default = "<AREA SHAPE=DEFAULT HREF=\"$1\" ALT=\"DEFAULT\">\n";
            }
            return;
        }
        elsif ($line =~ m/^\s*(circle|poly|rectangle)\s+([^\000]+)\s+(\S+)\s*$/i) {
            $type = $1;
            $href = $3;
            $coords = $2;
            $type =~ s/rectangle/rect/i; # CERN uses rectangle but we want rect
        }
        elsif ($line =~ m/^\s*#\s+(.+)\s*$/i) {
            $lastc = $1;
            print OUT "<!-- $1 -->\n";
        }
        else {
            chop($line);
            print OUT "<!-- Unrecognized line: $line -->\n";
            $lastc = "";
        }
    } elsif ($server =~ m|ncsa|i) {
        if ($line =~ m|^\s*default\s+(\S+)\s*$|i ) {
            if ($defshape eq "rect") {
                $default = "<AREA SHAPE=RECT COORDS=\"0,0,9999,9999\" HREF=\"$1\" ALT=\"DEFAULT\">\n";
            }
            else {
                $default = "<AREA SHAPE=DEFAULT HREF=\"$1\" ALT=\"DEFAULT\">\n";
            }
            return;
        }
        elsif ($line =~ m/^\s*(circle|poly|rect|point)\s+(\S+)\s+([^\000]+)$/i) {
            $type = $1;
            $href = $2;
            $coords = $3;
        }
        elsif ($line =~ m/^\s*#\s+(.+)\s*$/i) {
            $lastc = $1;
            print OUT "<!-- $1 -->\n";
        }
        else {
            chop($line);
            print OUT "<!-- Unrecognized line: $line -->\n";
            $lastc = "";
        }
    }
    else {
        print "Dodgy server set!\n";
        exit;
    }

    if ($type =~ m/(circle|poly|rect)/i) {
        # convert the coords to a comma separated list of numbers
        @coords = ();
        $type =~ tr|a-z|A-Z|;
        while($coords =~ s|^\D*(\d+)||) {
            push(@coords, $1);
        }
        $coords = join(",", @coords);
        if ($lastc ne "") {
            $alt = $lastc;
        }
        print OUT "<AREA SHAPE=$type COORDS=\"$coords\" HREF=\"$href\" ALT=\"$alt\">\n";
        $lastc = "";
        $alt = "";
    }
}
    
##EOF##
